src/compiler/parser.ts

```diff
@@ -666,6 +668,9 @@ namespace ts {
             sourceFile.languageVariant = getLanguageVariant(sourceFile.fileName);
             sourceFile.isDeclarationFile = fileExtensionIs(sourceFile.fileName, ".d.ts");
 
+            scriptKind = scriptKind || ScriptKind.Unknown;

```

Comment:
that feels wrong - I think that `parseSourceFile` should do this check and always provide `scriptKind` so at this point it always have concrete value